app/src/page/splash/view.js

```diff
@@ -88,37 +90,36 @@ class SplashScreen extends React.PureComponent {
       return;
     }
 
-    const blockchainStatus = status.blockchain_status;
-    if (blockchainStatus) {
+    const blockchainHeaders = status.blockchain_headers;
+    const walletStatus = status.wallet;
+
+    if (blockchainHeaders) {
       this.setState({
-        isDownloadingHeaders: blockchainStatus.is_downloading_headers,
-        headersDownloadProgress: blockchainStatus.headers_download_progress
+        isDownloadingHeaders: blockchainHeaders.downloading_headers,
+        headersDownloadProgress: blockchainHeaders.download_progress
       });
     }
 
-    if (blockchainStatus && (blockchainStatus.is_downloading_headers ||
-      (this.state.didDownloadHeaders && 'loading_wallet' === startupStatus.code))) {
+    if (blockchainHeaders && blockchainHeaders.downloading_headers && blockchainHeaders.download_progress < 100) {
       if (!this.state.didDownloadHeaders) {
         this.setState({ didDownloadHeaders: true });
```

Comment:
Aren't the headers not downloaded at this point?
`blockchainHeaders.download_progress < 100`
Then `didDownloadHeaders` is set to `true`

Reply:
I'm supposed to remove this `setState` call as it's not needed anymore. The check is also supposed to be `if (blockchainHeaders && blockchainHeaders.downloading_headers)`.